src/trainer_hill.c

```diff
@@ -0,0 +1,1121 @@
+}
+
+static void sub_81D59D0(void)
+{
+    if (gSaveBlock1Ptr->trainerHill.field_3D6E_0b)
+    {
+        gSpecialVar_Result = 2;
+    }
+    else if (GetTimerValue(&gSaveBlock1Ptr->trainerHill.field_3D68) > gSaveBlock1Ptr->trainerHill.field_3D64)
+    {
+        SetTimerValue(&gSaveBlock1Ptr->trainerHill.field_3D68, gSaveBlock1Ptr->trainerHill.field_3D64);
+        gSaveBlock1Ptr->trainerHillTimes[gSaveBlock1Ptr->trainerHill.tag] = gSaveBlock1Ptr->trainerHill.field_3D68;
+        gSpecialVar_Result = 0;
+    }
+    else
+    {
+        gSpecialVar_Result = 1;
+    }
+
+    gSaveBlock1Ptr->trainerHill.field_3D6E_0b = 1;
+}
+
+static void sub_81D5A70(void)
+{
+    if (!gSaveBlock1Ptr->trainerHill.field_3D6E_0c)
+    {
+        if (gSaveBlock1Ptr->trainerHill.field_3D64 >= HILL_MAX_TIME)
+            gSaveBlock1Ptr->trainerHill.field_3D64 = HILL_MAX_TIME;
+        else
+            sub_80008DC(&gSaveBlock1Ptr->trainerHill.field_3D64);
+    }
+}
+
+static void sub_81D5AB4(void)
+{
+    gSaveBlock1Ptr->trainerHill.field_3D6E_0d = 1;
+}
+
+static void sub_81D5AD0(void)
+{
+    if (gSaveBlock1Ptr->trainerHill.field_3D6E_0d)
+    {
+        gSaveBlock1Ptr->trainerHill.field_3D6E_0d = 0;
+        gSpecialVar_Result = 0;
+    }
+    else if (gSaveBlock1Ptr->trainerHill.field_3D6E_0e)
+    {
+        gSaveBlock1Ptr->trainerHill.field_3D6E_0e = 0;
+        gSpecialVar_Result = 1;
+    }
+    else
+    {
+        gSpecialVar_Result = 2;
+    }
+}
+
+static void sub_81D5B2C(void)
+{
+    s32 total, minutes, secondsWhole, secondsFraction;
+
+    total = gSaveBlock1Ptr->trainerHill.field_3D64;
+    if (total >= HILL_MAX_TIME)
+        total = HILL_MAX_TIME;
+
+    minutes = total / (60 * 60);
+    total %= (60 * 60);
+    secondsWhole = total / 60;
+    total %= 60;
+    secondsFraction = (total * 168) / 100;
+
+    ConvertIntToDecimalStringN(gStringVar1, minutes, STR_CONV_MODE_RIGHT_ALIGN, 2);
+    ConvertIntToDecimalStringN(gStringVar2, secondsWhole, STR_CONV_MODE_RIGHT_ALIGN, 2);
+    ConvertIntToDecimalStringN(gStringVar3, secondsFraction, STR_CONV_MODE_LEADING_ZEROS, 2);
+}
+
+static void sub_81D5BBC(void)
+{
+    SetUpDataStruct();
+    if (sHillData->tag.unkField_2 != 4)
+    {
+        ConvertIntToDecimalStringN(gStringVar1, sHillData->tag.unkField_2, STR_CONV_MODE_LEFT_ALIGN, 1);
+        gSpecialVar_Result = 0;
+    }
+    else
+    {
+        gSpecialVar_Result = 1;
+    }
+
+    FreeDataStruct();
+}
+
+static void sub_81D5C00(void)
+{
+    SetUpDataStruct();
+    gSpecialVar_Result = 0;
+    FreeDataStruct();
+}
+
+bool8 sub_81D5C18(void)
+{
+    if (VarGet(VAR_0x40D6) == 0)
+        return FALSE;
+    else if (gSaveBlock1Ptr->trainerHill.field_3D6E_0c)
+        return FALSE;
+    else if (GetCurrentTrainerHillMapId() != 0)
+        return TRUE;
+    else
+        return FALSE;
+}
+
+static void sub_81D5C5C(void)
+{
+    if (!sub_81D5C18())
+        gSpecialVar_Result = 0;
+    else
+        gSpecialVar_Result = 1;
+}
+
+void nullsub_129(void)
```

Comment:
Why is one nullsub static and the other one not?

Reply:
Because the other one is unused.